src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceImpl.java

```diff
@@ -76,4 +81,33 @@ public boolean validateUserPersistedWithException(HttpStatus status) {
         }
         return true;
     }
+
+    @Override
+    public boolean validateUserStatusWithException(UserProfile userProfile, IdamStatus expectedStatus) {
+        if (expectedStatus != userProfile.getStatus()) {
+            auditService.persistAudit(HttpStatus.BAD_REQUEST, ResponseSource.API);
+            final String exceptionMsg = String.format("User is not in %s state", expectedStatus);
+            exceptionService.throwCustomRuntimeException(ExceptionType.BADREQUEST, exceptionMsg);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean validateUserLastUpdatedWithinSpecifiedTimeWithException(UserProfile userProfile, long expectedMins) {
+        LocalDateTime latUpdated = userProfile.getLastUpdated();
+        if (Duration.between(latUpdated, LocalDateTime.now()).toMinutes() < expectedMins) {
+            auditService.persistAudit(HttpStatus.TOO_MANY_REQUESTS, ResponseSource.API);
+            final String exceptionMsg = String.format(TOO_MANY_REQUEST.getErrorMessage());
```

Comment:
please make it void instead of boolean if you are not adding any logic based on return value.

Reply:
done